editor/src/components/editor/component-button.tsx

```diff
@@ -39,39 +46,46 @@ export const ComponentOrInstanceIndicator = betterReactMemo('ComponentOrInstance
     [setIsOpen],
   )
 
-  enum editContext {
-    ELEMENT = 0,
-    COMPONENT_INSTANCE,
-    COMPONENT,
-    COMPONENT_CHILD,
-  }
+  const target = selectedViews[0]
 
-  const [currentEditContext, setCurrentEditContext] = React.useState(editContext.COMPONENT)
+  const isFocused = React.useCallback(() => {
+    if (target != null) {
+      return EP.isFocused(focusedElementPath, target)
+    } else {
+      return false
+    }
+  }, [focusedElementPath, target])
```

Comment:
Based on how it is used right now, it seems like `ifFocused` does not need to be a callback function, you might as well just save the value into a const and use that:

```suggestion
  const isFocused = target == null ? false : EP.isFocused(focusedElementPath, target)
```

beware that you also need to replace the usage from `isFocused()` to `isFocused` everywhere in the file!